### PR TITLE
address related updates

### LINKED ISSUE #63: Adding Address to Employee
Let's add address to the employee:

Fields to add:

AddressLine1
AddressLine2
City,
State,
ZipCode

Things to modify:

- Table definition
- Repository
- Services (create, update, select by id).
- Models

### PR DESCRIPTION
**Pull request by bwruan**

### CODE REVIEW

File: EmployeeManagementService/EmployeeManagementService.Domain/Services/EmployeeService.cs [hunk 1]
```diff
@@ -14,11 +14,13 @@ public interface IEmployeeService
         Task CreateEmployee(Employee newEmployee, string password);
         Task EmployeeLogIn(string username, string password);
         Task EmployeeLogout(long id);
+        Task<Employee> GetAddressByEmployeeId(long id);
```
Comment (jruan1580) [inline EmployeeManagementService/EmployeeManagementService.Domain/Services/EmployeeService.cs]: We don't really need a GetAddress or UpdateAddress.

This should just be done when we UpdateEmployee or CreateEmployee or GetEmployee

File: EmployeeManagementService/EmployeeManagementService.Domain/Models/Employee.cs [hunk 2]
```diff
@@ -32,6 +32,16 @@ public class Employee
         
         public bool? Active { get; set; }
 
+        public string AddressLine1 { get; set; }
```
Comment (jruan1580) [inline EmployeeManagementService/EmployeeManagementService.Domain/Models/Employee.cs]: Let's create an Address class in our Domain instead.

Add this new object to Employee and DTO map to an Address object

File: EmployeeManagementService/EmployeeManagementService.Domain/Mappers/Database/EmployeeMapper.cs [hunk 3]
```diff
@@ -20,6 +20,11 @@ public static CoreEmployee ToCoreEmployee(DbEmployee dbEmployee)
             coreEmp.TempPasswordChanged = dbEmployee.TempPasswordChanged;
             coreEmp.Status = dbEmployee.Status;
             coreEmp.Active = dbEmployee.Active;
+            coreEmp.AddressLine1 = dbEmployee.AddressLine1;
```
Comment (jruan1580) [inline EmployeeManagementService/EmployeeManagementService.Domain/Mappers/Database/EmployeeMapper.cs]: Map to and from address object instead.

File: EmployeeManagementService/EmployeeManagementService.Infrastructure/Persistence/EmployeeRepository.cs [hunk 4]
```diff
@@ -9,12 +9,14 @@ namespace EmployeeManagementService.Infrastructure.Persistence
 {
     public interface IEmployeeRepository
     {
-        Task CreateEmployee(string firstName, string lastName, string username, string ssn, byte[] password, string role, bool? active = true);
+        Task CreateEmployee(string firstName, string lastName, string username, string ssn, byte[] password, string role, string a1, string a2, string city, string state, string zip, bool? active = true);
```
Comment (jruan1580) [inline EmployeeManagementService/EmployeeManagementService.Infrastructure/Persistence/EmployeeRepository.cs]: We may need to change the way we do this in refactoring. Before we call create in service, we map domain to entity first and just pass in the entity.

File: EmployeeManagementService/EmployeeManagementService.Infrastructure/Persistence/EmployeeRepository.cs [hunk 5]
```diff
@@ -9,12 +9,14 @@ namespace EmployeeManagementService.Infrastructure.Persistence
 {
     public interface IEmployeeRepository
     {
-        Task CreateEmployee(string firstName, string lastName, string username, string ssn, byte[] password, string role, bool? active = true);
+        Task CreateEmployee(string firstName, string lastName, string username, string ssn, byte[] password, string role, string a1, string a2, string city, string state, string zip, bool? active = true);
+        Task<Employee> GetAddressByEmployeeId(long id);
```
Comment (jruan1580) [inline EmployeeManagementService/EmployeeManagementService.Infrastructure/Persistence/EmployeeRepository.cs]: No need for these. We can handle them all in GetEmployee, UpdateEmployee, and CreateEmployee

**Review by jruan1580 — COMMENTED** [review]
Also, add new address class in API and map them in employeeDTO

**Comment by bwruan** [comment]
addressed comments

**Pull request merged**